lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/ModifyingSuggester.java

```diff
@@ -30,9 +32,52 @@
   }
 
   LinkedHashSet<String> suggest(String word) {
+    tryVariationsOf(word);
+
+    WordCase wc = WordCase.caseOf(word);
+
+    if (wc == WordCase.MIXED) {
+      int dot = word.indexOf('.');
+      if (dot > 0
+          && dot < word.length() - 1
+          && WordCase.caseOf(word.substring(dot + 1)) == WordCase.TITLE) {
+        result.add(word.substring(0, dot + 1) + " " + word.substring(dot + 1));
+      }
+
+      tryVariationsOf(toLowerCase(word));
+    }
+
+    return result;
+  }
+
+  private String toLowerCase(String word) {
```

Comment:
I'm pretty sure we have a case conversion routine somewhere in the code (UnicodeUtil?)... Also, this does char-by-char conversion - it could be done on full unicode codepoints (which Java supports). I assume locale should not be taken into account here (to be identical to hunspell's behavior?). Don't know if this matters.

Reply:
`CharacterUtils` seems to have something similar. I haven't yet seen the need for code point support though, Hunspell uses UTF-16 internally which should fit Java strings unless I'm missing something.

This conversion is locale-dependent: it honors at least tr/az languages in `Dictionary.caseFold`.

Reply:
Ok, I looked at the entire code. Let's leave it for now - once we can pass all of hunspell's test cases perhaps there is a way to improve this. I'd love to see Java built-in locale mechanisms leveraged for doing some of the work here (like delegating letter case handling to Locale, replacing alternateCasing).